File: src/gametest/scala/dev/krysztal/casualtiesbelow/gametest/AchievementGameTests.scala
package dev.krysztal.casualtiesbelow.gametest

import net.minecraft.gametest.framework.GameTestHelper

import net.fabricmc.fabric.api.gametest.v1.GameTest

/** GameTest discovery entrypoint for the achievement scenarios ([[AchievementScenarios]]). */
final class AchievementGameTests {

  @GameTest(maxTicks = 20)
  def bandageObtainedGrantsAdvancement(helper: GameTestHelper): Unit = {
    AchievementScenarios.bandageObtainedGrantsAdvancement(helper)
  }

  @GameTest(maxTicks = 20)
  def refinedExtractObtainedGrantsAdvancement(helper: GameTestHelper): Unit = {
    AchievementScenarios.refinedExtractObtainedGrantsAdvancement(helper)
  }

  @GameTest(maxTicks = 20)
  def opioidOverdoseDeathGrantsAdvancement(helper: GameTestHelper): Unit = {
    AchievementScenarios.opioidOverdoseDeathGrantsAdvancement(helper)
  }

  @GameTest(maxTicks = 20)
  def hypoxiaDeathWithoutOpioidDoesNotGrant(helper: GameTestHelper): Unit = {
    AchievementScenarios.hypoxiaDeathWithoutOpioidDoesNotGrant(helper)
  }

  @GameTest(maxTicks = 20)
  def consciousnessMinimumGrantsAdvancement(helper: GameTestHelper): Unit = {
    AchievementScenarios.consciousnessMinimumGrantsAdvancement(helper)
  }

  @GameTest(maxTicks = 20)
  def consciousPlayerDoesNotGrantMinimum(helper: GameTestHelper): Unit = {
    AchievementScenarios.consciousPlayerDoesNotGrantMinimum(helper)
  }

  @GameTest(maxTicks = 20)
  def maxDiscomfortFoodGrantsAdvancement(helper: GameTestHelper): Unit = {
    AchievementScenarios.maxDiscomfortFoodGrantsAdvancement(helper)
  }

  @GameTest(maxTicks = 20)
  def ordinaryFoodDoesNotGrantMaxDiscomfort(helper: GameTestHelper): Unit = {
    AchievementScenarios.ordinaryFoodDoesNotGrantMaxDiscomfort(helper)
  }

  @GameTest(maxTicks = 20)
  def firstBleedingGrantsRoot(helper: GameTestHelper): Unit = {
    AchievementScenarios.firstBleedingGrantsRoot(helper)
  }

  @GameTest(maxTicks = 20)
  def hemostasisAfterNearMaxBleedingGrantsAdvancement(helper: GameTestHelper): Unit = {
    AchievementScenarios.hemostasisAfterNearMaxBleedingGrantsAdvancement(helper)
  }

  @GameTest(maxTicks = 20)
  def bleedingBelowThresholdDoesNotGrantHemostasis(helper: GameTestHelper): Unit = {
    AchievementScenarios.bleedingBelowThresholdDoesNotGrantHemostasis(helper)
  }
}

File: src/gametest/scala/dev/krysztal/casualtiesbelow/gametest/AchievementGameTests.java
package dev.krysztal.casualtiesbelow.gametest;

import net.fabricmc.fabric.api.gametest.v1.GameTest;
import net.minecraft.gametest.framework.GameTestHelper;

/** Java discovery bridge for the Scala achievement GameTest scenarios. */
public final class AchievementGameTests {
    @GameTest(maxTicks = 20)
    public void bandageObtainedGrantsAdvancement(GameTestHelper helper) {
        AchievementScenarios.bandageObtainedGrantsAdvancement(helper);
    }

    @GameTest(maxTicks = 20)
    public void refinedExtractObtainedGrantsAdvancement(GameTestHelper helper) {
        AchievementScenarios.refinedExtractObtainedGrantsAdvancement(helper);
    }

    @GameTest(maxTicks = 20)
    public void opioidOverdoseDeathGrantsAdvancement(GameTestHelper helper) {
        AchievementScenarios.opioidOverdoseDeathGrantsAdvancement(helper);
    }

    @GameTest(maxTicks = 20)
    public void hypoxiaDeathWithoutOpioidDoesNotGrant(GameTestHelper helper) {
        AchievementScenarios.hypoxiaDeathWithoutOpioidDoesNotGrant(helper);
    }

    @GameTest(maxTicks = 20)
    public void consciousnessMinimumGrantsAdvancement(GameTestHelper helper) {
        AchievementScenarios.consciousnessMinimumGrantsAdvancement(helper);
    }

    @GameTest(maxTicks = 20)
    public void consciousPlayerDoesNotGrantMinimum(GameTestHelper helper) {
        AchievementScenarios.consciousPlayerDoesNotGrantMinimum(helper);
    }

    @GameTest(maxTicks = 20)
    public void maxDiscomfortFoodGrantsAdvancement(GameTestHelper helper) {
        AchievementScenarios.maxDiscomfortFoodGrantsAdvancement(helper);
    }

    @GameTest(maxTicks = 20)
    public void ordinaryFoodDoesNotGrantMaxDiscomfort(GameTestHelper helper) {
        AchievementScenarios.ordinaryFoodDoesNotGrantMaxDiscomfort(helper);
    }

    @GameTest(maxTicks = 20)
    public void firstBleedingGrantsRoot(GameTestHelper helper) {
        AchievementScenarios.firstBleedingGrantsRoot(helper);
    }

    @GameTest(maxTicks = 20)
    public void hemostasisAfterNearMaxBleedingGrantsAdvancement(GameTestHelper helper) {
        AchievementScenarios.hemostasisAfterNearMaxBleedingGrantsAdvancement(helper);
    }

    @GameTest(maxTicks = 20)
    public void bleedingBelowThresholdDoesNotGrantHemostasis(GameTestHelper helper) {
        AchievementScenarios.bleedingBelowThresholdDoesNotGrantHemostasis(helper);
    }
}
